src/cli/log.rs

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use structopt::StructOpt;
+
+/// Common logging / output options
+#[derive(StructOpt, Debug)]
+pub struct LogOptions {
+    /// Increase log verbosity. Pass multiple times for more log output.
+    ///
+    /// By default we only show error messages. Passing `-v` will show warnings,
+    /// `-vv` adds info, `-vvv` for debug, and `-vvvv` for trace.
+    #[structopt(long = "verbose", short = "v", parse(from_occurrences))]
+    verbosity: u8,
+
+    /// Set fancier logging filters.
+    ///
+    /// This is equivalent to the `RUST_LOG` environment variable.
+    /// For more info, see the `env_logger` crate documentation.
+    #[structopt(long = "log-filter", env = "ENARX_LOG")]
+    log_filter: Option<String>,
+
+    /// Set log output target ("stderr", "stdout")
+    #[structopt(long, default_value = "stderr")]
+    log_target: LogTarget,
+}
+
+/// Represents targets for debug logging.
+/// This will probably grow over time.
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
+enum LogTarget {
+    Stdout,
+    Stderr,
+    // FUTURE: file path, syslog/journal, ...
```

Comment:
Just a question for my own understanding, how is `FUTURE` different from `TODO`?

Reply:
Heh, yeah, it is kind of a fuzzy difference. Here's how I use `FUTURE`, `TODO`, and `FIXME`:

I use `FUTURE` for possible future features - ideas that might be good or interesting, but wouldn't be trivial to add, aren't required by any specification, aren't on the roadmap, and aren't a blocker for release.

I use `TODO` when I'm pointing out known-missing/incomplete parts of _working_ code - it's not _broken_, but there's a feature/enhancement we know we'll need later. It's for things that are on the roadmap, or part of some spec, or for tech debt we know we'll need to repay later. Still not a release blocker, but shouldn't be thrown out lightly.

And then `FIXME` is for when I'm pointing out something that I know is just plain wrong, or broken, or silly. Those _should_ block release, except in special circumstances where we have a clear explanation of why we're leaving known-broken code in place, and there's a plan for how and when we're going to fix it.

...we should maybe put something in `STYLE.md` about this.